workflow/controller/workflowpod.go

```diff
@@ -696,12 +696,12 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template) {
```

Comment:
Can you add a test?

Reply:
Added tests for setting default and that those defaults are overriden by setting in workflow